pandas/core/groupby/groupby.py

```diff
@@ -663,7 +665,17 @@ def get_converter(s):
             converter = get_converter(index_sample)
             names = (converter(name) for name in names)
 
-        return [self.indices.get(name, []) for name in names]
+        # return [self.indices.get(name, []) for name in names]
+        # self.indices is a dict and doesn't handle looking up nulls in the groups
+        from pandas import (
+            Index,
+            Series,
+        )
+
+        index = Index(self.indices.keys(), tupleize_cols=False)
```

Comment:
can you add a comment about why this is two lines instead of just` indices = Series(self.indices)`?  i assume its to avoid a MultiIndex